File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        if(haystack.Length < needle.Length) return -1;

        var count = -1;
        for(int i = 0; i < haystack.Length; i++)
            if(haystack[i] == needle[++count] && count == needle.Length - 1)
                return i - count;
            else if(haystack[i] != needle[count])
            {
                i = i - count;
                count = -1;
            }

        return -1; 
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        
        int start = 0;
        
        for(int i=0; i<haystack.size(); i++) {
            
            if(haystack[i] == needle[start]) {
                start++;
            } else {
                i = i - start;
                start = 0;
            }
            
            if(start == needle.size())
                return i - start + 1;
        }
        
        return -1;
        
    }
};